src/transport/transport.cc

```diff
@@ -168,30 +169,41 @@ ConvectionTransport::~ConvectionTransport()
     unsigned int sbi;
 
     //Destroy mpi vectors at first
-    VecDestroy(&v_sources_corr);
     MatDestroy(&tm);
-    
-    VecDestroy(vconc);
-    VecDestroy(bcvcorr);
-    VecDestroy(vpconc);
-    VecDestroy(vcumulative_corr);
-    
+    VecDestroy(&vcfl_flow_);
```

Comment:
Try to use VectorSeqDouble in fields/vec_seq_double.hh for all internal vectors namely if both PETSc vector and raw ptr are necessary.